core/src/main/java/com/linecorp/armeria/common/metric/DropwizardMeterRegistry.java

```diff
@@ -0,0 +1,79 @@
+/*
+ *  Copyright 2017 LINE Corporation
+ *
+ *  LINE Corporation licenses this file to you under the Apache License,
+ *  version 2.0 (the "License"); you may not use this file except in compliance
+ *  with the License. You may obtain a copy of the License at:
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package com.linecorp.armeria.common.metric;
+
+import static java.util.Comparator.comparing;
+import static java.util.Objects.requireNonNull;
+
+import java.util.Iterator;
+
+import com.google.common.collect.Streams;
+
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.NamingConvention;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.util.HierarchicalNameMapper;
+
+/**
+ * {@link MeterRegistry} implementation for <a href="http://metrics.dropwizard.io/">Dropwizard Metrics</a>.
+ * This implementation adds more convenient constructors and sets more sensible default {@link NamingConvention}
+ * and {@link HierarchicalNameMapper} on top of the upstream implementation.
+ */
+@SuppressWarnings("ClassNameSameAsAncestorName")
+public class DropwizardMeterRegistry extends io.micrometer.core.instrument.dropwizard.DropwizardMeterRegistry {
```

Comment:
Didn't want to have a factory class for each registry type, because otherwise a user has to pull in both dropwizard and prometheus even if s/he doesn't use both.